src/unix/tty.c

```diff
@@ -62,7 +65,28 @@ int uv_tty_init(uv_loop_t* loop, uv_tty_t* tty, int fd, int readable) {
    * other processes.
    */
   if (type == UV_TTY) {
-    r = uv__open_cloexec("/dev/tty", O_RDWR);
+    /* Try to find a tty in stdin, stdout, and stderr. If a tty is found compare
+     * it to the fd. Only reopen the file descriptor if its inode matches the one
+     * bound to stdin, stdout, or stderr.
+     */
+    r = fstat(fd, &fdstat);
+    if (r < 0) {
+      /* Unlikely to happen, as a fstat() call has already happened in
+       * uv_guess_handle(). Nevertheless, another thread may have closed
+       * the fd so check this.
+       */
+      return -errno;
+    }
+
+    for (fdtty = 0, r = -1; fdtty <= STDERR_FILENO && !isatty(fdtty); fdtty++) {
```

Comment:
@Gottox Perhaps I'm misunderstanding your comment... if `&& !isatty(fdtty)` is removed from the for loop condition, then node cannot redirect stdout or stderr to another tty. Confirmed on mac with node 5.9.0 sources.